Fix errors around permissions on objects due to operator-sdk version bump metrics change

laurafitzgerald opened the pull request:
# Description
https://issues.redhat.com/browse/MGDAPI-326

## Verification

On a clean cluster - please uninstall any previous installation and delete any integreatly-operator namespaces.
In the Makefile ->
Update `NAMESPACE_PREFIX=redhat-mgdapi-`
Update `export INSTALLATION_TYPE   ?= managed-api`
Update `export INSTALLATION_PREFIX ?= redhat-managed-api`
Run `make cluster/prepare/local`
Run `yq w -i deploy/operator.yaml 'spec.template.spec.containers[0].image' quay.io/laurafitzgerald/integreatly-operator:v2.8.6`
Run `oc apply -f deploy/operator.yaml -n redhat-mgdapi-operator`

Verify that the errors faced in [this gist ](https://gist.github.com/psturc/86df94fd4fac09c3d32a3e94e3c7d806)are no longer present.

Please go to prometheus and ensure no alerts are firing. There's a change include which fixes an issue with alert manager.


## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist
- [ ] This change requires a documentation update <!-- Update JIRA with Affects -> Documentation -->
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

laurafitzgerald commented on cmd/manager/main.go:
```diff
@@ -293,3 +305,24 @@ func setupWebhooks(mgr manager.Manager) error {
 	return nil
 }
+
+func isKubeMetaKind(kind string) bool {
```
This list is lifted from here: https://github.com/operator-framework/operator-sdk/blob/v0.15.2/pkg/k8sutil/k8sutil.go#L142-L180

laurafitzgerald commented on pkg/controller/installation/installation_controller.go:
```diff
@@ -251,7 +251,7 @@ func (r *ReconcileInstallation) Reconcile(request reconcile.Request) (reconcile.
 		}
 	}
-
+	metrics.SetRateLimiting(20)
```
this is a WIP: the value will be picked up from the config map created by: https://github.com/integr8ly/integreatly-operator/pull/1234

laurafitzgerald commented on pkg/products/monitoring/reconciler.go:
```diff
@@ -639,7 +639,7 @@ func (r *Reconciler) reconcileAlertManagerConfigSecret(ctx context.Context, serv
 	smtpSecret := &corev1.Secret{}
 	if err := serverClient.Get(ctx, types.NamespacedName{Name: r.installation.Spec.SMTPSecret, Namespace: rhmiOperatorNs}, smtpSecret); err != nil {
-		logrus.Warnf("could not obtain smtp credentials secret: %v", err)
+		return integreatlyv1alpha1.PhaseFailed, err
```
This change is required as if the secret is not found the monitoring installation continues but the config for alert manager is built incorrectly without error in integreatly-operator but the incorrect config causes the alert manager pod to error out.

KevFan replied:
So would the monitoring stage still complete if there is no smtp secret? :thinking: Just wondering cause I think work was done to allow installation to continue even if the smtp secret is not found and this would block installation at the monitoring stage :thinking: 
* https://github.com/integr8ly/integreatly-operator/pull/1160
* https://github.com/integr8ly/integreatly-operator/pull/865

psturc replied:
Good point @KevFan, I haven't tested it without smtp secret being present.

@philbrookes @pmccarthy wdyt?

laurafitzgerald replied:
@KevFan no it wouldn't complete. But it's not obvious that there is some problem with the installation and the alert manager pod errors without any feedback to the user/installer.

laurafitzgerald replied:
This code change is removed. Thanks for spotting this @KevFan 👍

laurafitzgerald commented:
/hold

psturc commented:
@laurafitzgerald I was able to verify the installation, then I hit the issue with uninstallation and saw this in rhmi-operator log:
```
time="2020-10-19T17:02:50Z" level=info msg="Uninstalling 3scale in stage uninstall - products"
E1019 17:02:51.129420       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:51.417070       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:51.418218       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.APIManager: apimanagers.apps.3scale.net is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "apimanagers" in API group "apps.3scale.net" in the namespace "redhat-mgdapi-3scale"
E1019 17:02:51.419275       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:51.420325       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:51.421477       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:51.422431       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:51.423577       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:51.424627       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:51.425709       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:51.426785       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:52.131434       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:52.419014       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:52.420006       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.APIManager: apimanagers.apps.3scale.net is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "apimanagers" in API group "apps.3scale.net" in the namespace "redhat-mgdapi-3scale"
E1019 17:02:52.421038       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:52.422059       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:52.423235       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:52.424253       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:52.425166       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:52.426221       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:52.427358       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:52.428370       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:53.133436       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:53.421006       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:53.421921       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.APIManager: apimanagers.apps.3scale.net is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "apimanagers" in API group "apps.3scale.net" in the namespace "redhat-mgdapi-3scale"
E1019 17:02:53.423135       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakBackup: keycloakbackups.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakbackups" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:53.423842       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:53.425146       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:53.425879       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakClient: keycloakclients.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakclients" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:53.427225       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.Keycloak: keycloaks.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloaks" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:53.428040       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-user-sso"
E1019 17:02:53.429114       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakRealm: keycloakrealms.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakrealms" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
E1019 17:02:53.430175       1 reflector.go:153] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:136: Failed to list *v1alpha1.KeycloakUser: keycloakusers.keycloak.org is forbidden: User "system:serviceaccount:redhat-mgdapi-operator:rhmi-operator" cannot list resource "keycloakusers" in API group "keycloak.org" in the namespace "redhat-mgdapi-rhsso"
time="2020-10-19T17:02:53Z" level=info msg="Reconciling 3scale"
```

Issue with uninstallation: https://issues.redhat.com/browse/MGDAPI-404

Not sure if it might be related?

laurafitzgerald commented:
/test rhoam-e2e

laurafitzgerald commented:
@psturc I cleaned up the pr as we discussed. 
Thanks for the easier instructions for installing it in cluster.
I've dropped the extra commit and installed this on my cluster and it works as documented in the description.

I'm still investigating the uninstall issue that you faced. I'm not sure if that's an existing issue or something new created.

psturc commented:
Verified
/lgtm
/approve

laurafitzgerald commented:
/unhold

laurafitzgerald commented:
/test rhoam-e2e

laurafitzgerald commented:
/test e2e

laurafitzgerald commented:
/test e2e

laurafitzgerald commented:
/test e2e

KevFan commented:
/lgtm

pmccarthy commented:
/approve
